zebra-state/src/service/memory_state.rs

```diff
@@ -476,6 +620,14 @@ mod tests {
     use self::assert_eq;
     use super::*;
 
+    struct NoDebug<T>(T);
+
+    impl<T> fmt::Debug for NoDebug<Vec<T>> {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            write!(f, "{}, len={}", std::any::type_name::<T>(), self.0.len())
+        }
+    }
```

Comment:
Is this a shorter debug?

Reply:
yea, without this the output on test failure is completely unmanageable, just thousands upon thousands of lines of output for one variable.

Reply:
Can I suggest renaming it?

It doesn't actually disable debug output, so `NoDebug` is confusing.

```suggestion
    struct SummaryDebug<T>(T);

    impl<T> fmt::Debug for SummaryDebug<Vec<T>> {
        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
            write!(f, "{}, len={}", std::any::type_name::<T>(), self.0.len())
        }
    }
```

Reply:
Also, would it help to show an example first and last item?
Or are the items also too long?

Reply:
The items are waaay too long